lib/components/normal-components/PinHeader.ts

```diff
@@ -2,9 +2,17 @@ import { pinHeaderProps, type SchematicPortArrangement } from "@tscircuit/props"
 import { NormalComponent } from "../base-components/NormalComponent/NormalComponent"
 import { Port } from "../primitive-components/Port"
 import type { SourceSimplePinHeader } from "circuit-json"
-import type { BaseSymbolName } from "lib/utils/constants"
+import type { z } from "zod"
 
 export class PinHeader extends NormalComponent<typeof pinHeaderProps> {
+  constructor(props: z.input<typeof pinHeaderProps>) {
+    const propsWithDefaults = {
+      ...props,
+      facingDirection: props.facingDirection ?? "right",
+    }
+    super(propsWithDefaults)
```

Comment:
don't modify props (props are immutable)

Instead, create a function called `_getFacingDirection()` and force anything using facing direction to use that. This allows much cleaner overrides and preserves the user's definitions (which makes errors and lots of other stuff more clear)